6-25-19-smirnoff99Frost-coverage/create_optimization_dataset.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+
+import copy
+import json
+from collections import Counter
+import qcportal as ptl
+from qcelemental.models import Molecule
+
+def read_molecules(input_json):
+    """ Extract the molecules and the index of them from the input json file
+    Return a dictionary like
+    {
+        index1: Molecule1,
+        index2: Molecule2,
+    }
+
+    Note
+    ----
+    1. The mdata['cmiles_identifiers']['canonical_isomeric_smiles'] is selected as the index.
+    2. For molecules have the same "canonical_isomeric_smiles", we use index-1, index-2 to distinguish them.
+    """
+    res = {}
+    with open(input_json) as infile:
+        molecule_data_list = json.load(infile)
+    index_counter = Counter()
+    for mdata in molecule_data_list:
+        initial_molecules = mdata['initial_molecules']
+        cmiles_ids = mdata['cmiles_identifiers']
+        index = cmiles_ids['canonical_isomeric_smiles']
+        for i_conformer, initial_molecule in enumerate(initial_molecules):
+            qcel_molecule = Molecule.from_data(initial_molecule)
+            # use count to generate unique index
+            index_count = index_counter[index]
+            this_index = f'{index}-{index_count}'
+            index_counter[index] += 1
+            assert this_index not in res, f"Multiple molecules have the same index, please check {mdata}"
+            res[this_index] = qcel_molecule
+    return res
+
+
+def create_optimization_dataset(molecules_dict, client_config_file, dataset_name, qm_method, qm_basis, spec_name='default', start_compute=False):
+    """
+    Submit a list of molecules for optimization
+
+    Parameters
+    ----------
+    molecules_dict: dict {str: qcelemental.models.Molecule}
+        The dictionary contains molecules to be submitted. The key is the index (name) of the molecule, value is a Molecule object.
+    client_config_file: str
+        File name for the QCPortal client configuration.
+    dataset_name: str
+        Name of the dataset to be created
+    qm_method: str
+        QM method to use for geometry optimization
+    qm_basis: str
+        QM basis set to use for geometry optimization
+    spec_name: str
+        The name of the QM spec, default is "default"
+    start_compute: bool
+        If true, start compute this dataset after creation
+
+    Returns
+    -------
+    ds: qcportal.collections.OptimizationDataset
+        The dataset created
+
+    """
+    client = ptl.FractalClient.from_file(client_config_file)
+    # create a new dataset with specified name
+    print(f"Creating OptimizationDataset < {dataset_name} >")
+    ds = ptl.collections.OptimizationDataset(dataset_name, client=client)
+    # create specification for this dataset
+    opt_spec = {"program": "geometric"}
```

Comment:
Do we need to specify other geometric options here for this dataset?

Reply:
I think the default options should be good for this free optimization with no constraints.